submit-api/src/submit_api/models/queries/project.py

```diff
@@ -13,24 +13,52 @@
 # limitations under the License.
 """Model to handle all complex operations related to User."""
 from submit_api.models import AccountProject, Project, db
+from submit_api.models.account_project_search_options import AccountProjectSearchOptions
+from submit_api.models.package import Package
+# pylint: disable=too-few-public-methods
 
 
-# pylint: disable=too-few-public-methods
 class ProjectQueries:
     """Query module for complex projects queries"""
 
     @classmethod
-    def get_projects_by_account_id(cls, account_id: int):
-        """Find projects by account_id"""
-        result = (
-            db.session.query(AccountProject)
-            .filter(AccountProject.account_id == account_id)
-            .join(Project)
-        )
-        return result
+    def get_projects_by_account_id(cls, account_id: int, search_options=AccountProjectSearchOptions):
+        """Find projects by account_id with optional search and pagination."""
+        query = db.session.query(AccountProject).filter(
+            AccountProject.account_id == account_id
+        ).join(Project)
+
+        # Apply search filters if provided
+        if search_options:
+            query = cls.apply_search_filters(query, search_options)
+
+        return query.all()
 
     @classmethod
     def get_projects_by_proponent_id(cls, proponent_id: int):
         """Find projects by proponent_id"""
-        result = db.session.query(Project).filter(Project.proponent_id == proponent_id)
-        return result
+        query = db.session.query(Project).filter(
+            Project.proponent_id == proponent_id
+        )
+        return query.all()
+
+    @classmethod
+    def apply_search_filters(cls, query, search_options):
+        """Apply various filters based on search options."""
+        if search_options.search_text:
+            query = cls._filter_by_search_text(
+                query, search_options.search_text)
+        if hasattr(search_options, 'status'):
+            query = cls._filter_by_status(query, search_options.status)
+        # Additional filters can be added here in the future
+        return query
+
+    @classmethod
+    def _filter_by_search_text(cls, query, search_text):
+        """Filter by search text across project name and package name."""
+        query = query.join(Package, AccountProject.packages).filter(
+            db.or_(
```

Comment:
what's the usage of the or operation here?

Reply:
im filtering through the submission names, (past & active submissions)

Reply:
sure, but why do u need the or?

- and: if condition A is true AND condition B is true return true
- or: if condition A is true OR condition B is true return true

why do u need it?

Reply:
small over sight on my end just removed that